enhancements/network/egress-ip-multi-nic.md

```diff
@@ -0,0 +1,227 @@
+---
+title: egress-ip-multi-nic
+authors:
+  - "@martinkennelly"
+reviewers:
+  - "@trozet"
+  - "@kyrtapz"
+  - "@flavio-fernandes"
+approvers:
+  - "@trozet"
+  - "@knobunc"
+api-approvers:
+- "@trozet"
+creation-date: 2023-05-16
+last-updated: 2023-05-30
+status: implementable
+tracking-link:
+- https://issues.redhat.com/browse/SDN-2097
+---
+# Egress IP Multi-NIC
+## Summary
+For OpenShift release version 4.13 or less, users may utilize egress IP to allow source NAT when packets leave the primary network.
+This enhancement will add egress IP support for networks that exists on NICs that aren't managed by OVN and hence forth
+these networks are called secondary networks.
+
+## Motivation
+Users may not wish to route workload traffic over OVN managed primary network and instead prefer to use a secondary network.
+With this enhancement, we can support segregation of workload traffic over multiple NICs while allowing a consistent source IP address.
+
+### Goals
+- Support egress IP for secondary networks
+
+### Non-Goals
+- Support egress IP on non-default networks controllers aka secondary network controllers
+- Support cloud egress IP assignment for secondary interfaces using cloud network config controller
+  - This maybe easy to do eventually, but it will require a lot of extra testing, so excluding it for now, and it's
+    a stretch goal.
+- Support egress IP on secondary bridges managed by OVN aka br-ex1
+
+## Proposal
+
+### Implementation Details/Notes/Constraints
+Users may provide not only a primary network but also additional networks that they want to utilize and support egress IP.
+Cluster manager will continue to provision `EgressIP` to nodes. Currently, it only assigns them to nodes which have a
+well-known label egress assignable `k8s.ovn.org/egress-assignable`.
+
+This enhancement is proposing enhancing cluster manager if the egress IP is within a subnet of a known non-primary network, then it
+will also consider which node hosts this secondary network. If no secondary network is found, we continue the default
+behaviour of assigning the egress IP to the primary network like what we do today.
```

Comment:
This is likely a misconfiguration, but it may be worth mentioning here:
the eip will be potentially on the primary network of all nodes, or on the secondary. We do not antecipate the eip being on either one across nodes. ---  or it that a non-issue and we would be supporting whichever, on a node to node basis?

Reply:
I removed this section - if `EgressIP` after learning today we make sure it falls within the primary network subnet. So, if egress IP doesnt find a network, we dont assign it to a node.